controllers/user_reconciler.go

```diff
@@ -0,0 +1,298 @@
+					log.Infof("Created  %s secret", htpasswdSecret.Name)
+				}
+			}
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// deleteUsers delete users in openshift cluster
+func (r *WorkshopReconciler) deleteUsers(workshop *workshopv1.Workshop) (reconcile.Result, error) {
+	var userList []string
+	userPrefix := workshop.Spec.UserDetails.UserNamePrefix
+	labelSelector, err := labels.Parse(UserLabelSelector)
+	if err != nil {
+		log.Errorf("Error %s", err)
+	}
+
+	ListUsers := &userv1.UserList{}
+	listOps := &client.ListOptions{
+		LabelSelector: labelSelector,
+	}
+	if err := r.List(context.TODO(), ListUsers, listOps); err != nil {
+		log.Errorf("Error %s", err)
+	}
+	for _, user := range ListUsers.Items {
+		username := user.Name
+		userList = append(userList, username)
+	}
+
+	createdUsers := len(userList)
+	for createdUsers >= 1 {
+		username := fmt.Sprint(userPrefix, createdUsers)
+		if result, err := r.deleteOpenshiftUser(workshop, r.Scheme, username); util.IsRequeued(result, err) {
+			return result, err
+		}
+		createdUsers--
+	}
+	if result, err := r.DeleteUserHtpasswd(workshop); err != nil {
+		return result, err
+	}
+
+	//Success
+	return reconcile.Result{}, nil
+}
+
+// deleteUser delete OpenShift user
+func (r *WorkshopReconciler) deleteOpenshiftUser(workshop *workshopv1.Workshop, scheme *runtime.Scheme, username string) (reconcile.Result, error) {
+
+	// Get user
+	userFound := &userv1.User{}
+	if err := r.Get(context.TODO(), types.NamespacedName{Name: username}, userFound); err != nil {
+		log.Errorf("Failed to find %s User", userFound.Name)
+	}
+	//
+	// Delete User Identity Mapping
+	userIdentity := openshiftuser.NewUserIdentityMapping(workshop, r.Scheme, USER_IDENTITY_MAPPING_NAME, username)
+	if err := r.Delete(context.TODO(), userIdentity); err != nil {
+		return reconcile.Result{}, err
+	}
+	log.Infof("Deleted %s User Identity Mapping ", userIdentity.Name)
+
+	// Delete Identity
+	identity := openshiftuser.NewIdentity(workshop, r.Scheme, username, IDENTITY_NAME, userFound)
+	if err := r.Delete(context.TODO(), identity); err != nil {
+		return reconcile.Result{}, err
+	}
+	log.Infof("Deleted %s Identity  ", identity.Name)
+
+	// Delete User Role Binding
+	userRoleBinding := openshiftuser.NewUserRoleBinding(workshop, r.Scheme, username, USER_ROLE_BINDIN_NAMESPACE_NAME,
+		USER_ROLE_BINDING_NAME, KIND_CLUSTER_ROLE)
+	if err := r.Delete(context.TODO(), userRoleBinding); err != nil {
+		return reconcile.Result{}, err
+	}
+	log.Infof("Deleted %s Role Binding", userRoleBinding.Name)
+
+	// Delete User
+	user := openshiftuser.NewUser(workshop, r.Scheme, username, userLabels)
+	if err := r.Delete(context.TODO(), user); err != nil {
+		return reconcile.Result{}, err
+	}
+	log.Infof("Deleted %s user", user.Name)
+
+	//Success
+	return reconcile.Result{}, nil
+}
+
+//DeleteUserHTPasswd delete Htpasswd secret for users
+func (r *WorkshopReconciler) DeleteUserHtpasswd(workshop *workshopv1.Workshop) (reconcile.Result, error) {
```

Comment:
@MaazMS why **D** is capital?

Reply:
@rasheedamir I have changed method name `DeleteUserHtpasswd` to  `deleteUserHtpasswd`